staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -385,6 +508,18 @@ func envelopePrefixTransformer(config *apiserverconfig.KMSConfiguration, envelop
 	}, nil
 }
 
+func envelopekmsv2PrefixTransformer(config *apiserverconfig.KMSConfiguration, envelopeService envelopekmsv2.Service, prefix []byte) (value.PrefixTransformer, error) {
+	// using AES-GCM by default for encrypting data with KMSv2
+	envelopeTransformer, err := envelopekmsv2.NewEnvelopeTransformer(envelopeService, int(*config.CacheSize), aestransformer.NewGCMTransformer, config.Name)
+	if err != nil {
+		return value.PrefixTransformer{}, err
+	}
+	return value.PrefixTransformer{
+		Transformer: envelopeTransformer,
+		Prefix:      prefix,
```

Comment:
this isn't including the config.Name in the prefix, so if there was more than one kmsv2 provider, all kmsv2 encoded data would attempt to be handled by all kmsv2 providers, right? should a test have noticed that conflict?

Reply:
@enj pointed out that the config.Name is encoded inside EncryptedObject as PluginName and the kmsv2 handler skips objects that don't match

I don't see a good reason to make all configured kmsv2 handlers decode each others' EncryptedObject and then skip... I'd recommend sticking with the existing prefix approach and dropping PluginName from EncryptedObject

Reply:
@enj also noticed we don't validate configured KMS names don't contain a colon... we probably should

Reply:
> I don't see a good reason to make all configured kmsv2 handlers decode each others' EncryptedObject and then skip... I'd recommend sticking with the existing prefix approach and dropping PluginName from EncryptedObject

Changed to using `k8s:enc:kms:v2:` for the prefix. 

> @enj also noticed we don't validate configured KMS names don't contain a colon... we probably should

@liggitt For this validation, does it makes sense to add the validation [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/apis/config/validation/validation.go) and fail if `invalid && apiVersion == v2`? Or should we do the validation in config.go?

Reply:
> @liggitt For this validation, does it makes sense to add the validation [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/apis/config/validation/validation.go) and fail if invalid && apiVersion == v2?

yeah, that's a good point at which to do the validation

Reply:
added the validation